cmd/geth/snapshot.go

```diff
@@ -149,6 +165,68 @@ It's also usable without snapshot enabled.
 	}
 )
 
+func pruneBlock(ctx *cli.Context) error {
+	stack, config := makeConfigNode(ctx)
+	defer stack.Close()
+
+	chaindb := utils.MakeChainDatabaseForBlockPrune(ctx, stack, false)
+	chaindb.Close()
+
+	var oldAncientPath, newAncientPath string
+	if path := getAncientPath(ctx); path != "" {
+		oldAncientPath = path + "/ancient"
+		newAncientPath = path + "/ancient_back"
+	} else {
+		utils.Fatalf("Prune failed, did not specify the AncientPath %v")
+		return errors.New("Prune failed, did not specify the AncientPath")
+	}
+
+	//TODO for filelock
+	//lock, _, err := fileutil.Flock(filepath.Join(oldAncientPath, "FLOCK"))
+	// lock, _, err := fileutil.Flock(oldAncientPath)
+	// if err != nil {
+	// 	return err
+	// }
+	// _, err = os.Open(oldAncientPath)
+	// if err != nil {
+	// 	utils.Fatalf("Failed to read genesis file: %v", err)
+	// }
+	for _, name := range []string{"chaindata"} {
+		root := stack.ResolvePath(name)
+		switch {
+		case oldAncientPath == "":
+			oldAncientPath = filepath.Join(root, "ancient")
+		case !filepath.IsAbs(oldAncientPath):
+			oldAncientPath = stack.ResolvePath(oldAncientPath)
+		}
+		pruner, err := pruner.NewBlockPruner(chaindb, stack, oldAncientPath)
+		if err != nil {
+			utils.Fatalf("Failed to create block pruner", err)
+			return err
+		}
+
+		if err := pruner.BlockPruneBackUp(name, config.Eth.DatabaseCache, utils.MakeDatabaseHandles(), newAncientPath, oldAncientPath, "", false); err != nil {
+			log.Error("Failed to back up block", "err", err)
+			return err
+		}
+	}
+
+	log.Info("geth block offline pruning backup successfully")
+
+	//After backing up successfully, rename the new ancientdb name to the original one, and delete the old ancientdb
+	if err := pruner.BlockPrune(oldAncientPath, newAncientPath); err != nil {
+		utils.Fatalf("Failed to prune block", err)
+		return err
+	}
+	//TODO lock.Release()
+	log.Info("Block prune successfully")
+	return nil
+}
+
+func getAncientPath(ctx *cli.Context) string {
```

Comment:
let us follow the same code style. Please remove this func, but use `ctx.GlobalString(utils.AncientFlag.Name)` directly.